clientdb/order.go

```diff
@@ -719,6 +743,12 @@ func deserializeOrderTlvData(r io.Reader, o order.Order) error {
 				return err
 			}
 		}
+
+		t, ok := parsedTypes[bidUnannoucedChannelsType]
+		if ok && t == nil {
```

Comment:
This check only makes sure we were able to parse the value correctly. But we still need to add `&& bidUnannoucedChannels == 1`.

Reply:
technically the value should be there ONLY IF it was true `bidUnannoucedChannels == 1` I added a comment but I guess that I can make it even clear checking for it anyway. :+1:

Reply:
Ah, I see, the whole record wouldn't be there so the `ok` wouldn't be true. Fair enough, but even clearer now I think.